auth/src/main/java/io/grpc/auth/GoogleAuthLibraryCallCredentials.java

```diff
@@ -56,6 +58,8 @@
  * Wraps {@link Credentials} as a {@link CallCredentials}.
  */
 final class GoogleAuthLibraryCallCredentials implements CallCredentials {
+  private static final String SERVICE_ACCOUNT_CLASS_NAME
+      = "com.google.auth.oauth2.ServiceAccountCredentials";
```

Comment:
> It is going to make compiling hard internally without the dep, or force the dep to be present even if unused.

I was just going to force the dep even when unused. That seems to be the Google/Blaze way. All other grpc implementations seem to be forcing the equivalent dep all the time.

> Please factor all uses of ServiceAccountCredentials to an individual Java file.

I don't see how that would help. We can't auto-select a file based on whether the class is present or not. That would really only help if we knew at compile-time whether ServiceAccountCredentials was present _and_ had separate targets.